src/windows/line.rs

```diff
@@ -0,0 +1,150 @@
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use pdb::{AddressMap, PdbInternalRva};
+use std::fmt::{Display, Formatter};
+
+#[derive(Clone, Debug)]
+pub struct Line {
+    pub rva: u32,
+    pub num: u32,
+    pub len: u32,
+    pub file_id: u32,
+}
+
+#[derive(Debug, Default)]
+pub struct Lines {
+    pub lines: Vec<Line>,
+    pub is_sorted: bool,
+}
+
+impl Display for Lines {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        for line in self.lines.iter() {
+            writeln!(
+                f,
+                "{:x} {:x} {} {}",
+                line.rva, line.len, line.num, line.file_id
+            )?;
+        }
+        Ok(())
+    }
+}
+
+impl Lines {
+    fn compute_len(&mut self, sym_len: u32) {
+        // The length (in the binary) of the line is not in the pdb but we can infer it:
+        // RVA     LINE NUMBER
+        // 0x0001  10  <= the size of line 10 is 0x000B - 0x0001
+        // 0x000B  11
+        // ...
+        // 0x002A  15 <= the size of line 15 is sym length - (0x002A - 0x0001)
+
+        if self.lines.is_empty() {
+            return;
+        }
+
+        if !self.is_sorted {
+            self.lines.sort_by_key(|x| x.rva);
+        }
+
+        let n_lines = self.lines.len();
+
+        unsafe {
+            let first_rva = if n_lines > 1 {
+                // The 2 next lines are valid because n_lines > 1
+                let next_rva = self.lines.get_unchecked(1).rva;
+                let line = self.lines.get_unchecked_mut(0);
+
+                let first_rva = line.rva;
+                line.len = next_rva - first_rva;
+
+                for i in 1..n_lines - 1 {
+                    // Valid because i + 1 < n_lines - 1 + 1
+                    let next_rva = self.lines.get_unchecked(i + 1).rva;
+                    let line = self.lines.get_unchecked_mut(i);
+                    line.len = next_rva - line.rva;
+                }
+
+                first_rva
+            } else {
+                // Valid because self.lines is not empty
+                self.lines.get_unchecked(0).rva
+            };
+
+            // Valid because self.lines is not empty
+            let last = self.lines.get_unchecked_mut(n_lines - 1);
+            last.len = sym_len - (last.rva - first_rva);
+        }
+    }
+
+    fn compute_rva(&mut self, address_map: &AddressMap) {
+        // the rva is the internal rva (needed to compute the length)
+        // So replace them (and eventually split lines) by the rva in the binary
```

Comment:
Can we define `rva` somewhere?  I assume it means "return value address", but I'm not sure about that, and I'm even less sure how it applies here.

Reply:
Done.
I added a comment in the struct Line.